pyo3-build-config/src/lib.rs

```diff
@@ -146,6 +146,11 @@ pub fn print_feature_cfgs() {
     if rustc_minor_version >= 59 {
         println!("cargo:rustc-cfg=thread_local_const_init");
     }
+
+    // Enable use of OnceLock on Rust 1.70 and greater
+    if rustc_minor_version >= 70 {
+        println!("cargo:rustc-cfg=once_lock");
```

Comment:
Can we please not do this? This code will be significantly simpler without conditional compilation, and I don't see the benefit.

Reply:
Hence, the suggestion to just use `AtomicI64` in https://github.com/PyO3/pyo3/pull/3446#discussion_r1321606949

Reply:
Yep I'm expecting to remove this later 😁